### PR TITLE
Update Snippy to version 3.2 with all new tests 

### PR DESCRIPTION
**Pull request by Slugger70**
Still haven't got a test for snippy-core as I'm struggling to find something small enough.

### CODE REVIEW

**Comment by bgruening** [comment]
@Slugger70 any idea why the test is running over and over again and fails?

**Comment by Slugger70** [comment]
Yes, it's still picking up the wrong build of freebayes.. It keeps getting the py36_1 build instead of the py36_2 build.. the _2 build only exists for linux and not for both mac and linux.. Would this be effecting which version gets selected?

File: tools/snippy/snippy-core.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<tool id="snippy-core" name="snippy-core" version="3.1">
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: Can you give this a proper name and description?

File: tools/snippy/snippy-core.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    <requirements>
+        <requirement type="package" version="3.1">snippy</requirement>
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: version number should be a macro? and maybe the entire requirement as well?

File: tools/snippy/snippy-core.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        perl "$__tool_directory__/snippy_core_wrapper.pl"
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: singel quotes

File: tools/snippy/snippy-core.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            "$is_reference"
+            --indirs '${" ".join(map(str, $indirs))}'
+
+    ]]></command>
+
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: I would remove these empty lines.

File: tools/snippy/snippy.xml [hunk 5]
```diff
@@ -0,0 +1,211 @@
+<tool id="snippy" name="snippy" version="3.1">
+    <requirements>
+        <requirement type="package" version="3.1">snippy</requirement>
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: macro, description of the tool is missing ...

File: tools/snippy/snippy.xml [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        #if $ref.is_of_type("fasta")
+            cp "$ref" "foo.fna" &&
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: single quotes

File: tools/snippy/snippy.xml [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+        #end if
+        #if $ref.is_of_type("genbank")
+            cp "$ref" "foo.gbk" &&
+        #end if
+        snippy
+            --outdir "out"
+            --cpus "\${GALAXY_SLOTS:-1}"
+            #if $ref.is_of_type("fasta")
+                --ref "foo.fna"
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: single quotes

File: tools/snippy/snippy-core.xml (hunk 1 above)
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: Should this be called - snippy merge?

File: tools/snippy/snippy-core.xml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        <data format="txt" name="alignment_summary" label="${tool.name} on ${on_string} core alignment summary" from_work_dir="core.txt" />
+    </outputs>
+    
+    <tests>
+      
+    </tests>
+
+    <help><![CDATA[
+**snippy-core 3.1**
+
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: Is the zip file really needed or could we also use the single files from snippy for this tool? My motivation is to get rid of the zip file, which does not have its own datatype and just clutters the history.

Comment (Slugger70) [inline tools/snippy/snippy-core.xml]: I asked Torsten about this and snippy-core is written to use the structure of the default snippy output directory. Including all of the subdirectory structures.. I'll see what I can figure out.

**Comment by bgruening** [comment]
ping @Slugger70

**Comment by Slugger70** [comment]
Hey Björn. I've asked the author about the zip files etc.. Snippy would need to be re-written to get around it. It's just an unfortunate consequence of how it works. I have another idea which involve removing the snippy core tool and replacing it with a tool that runs snippy on multiple inputs and then combines the outputs with snippy-core transparently to the user. Not sure what to do here. I can remove the PR until we sort it out if you like.

**Comment by bgruening** [comment]
@Slugger70 I don't want to block you here. If there is no way around the zip I fine with it for the time being.

**Comment by Slugger70** [comment]
I'll work some more on this tool for a bit and see what happens.. I think I've convinced the author to change some things.

**Comment by bgruening** [comment]
ping @Slugger70

**Comment by Slugger70** [comment]
I updated snippy to 3.2 as Torsten made some changes to make Galaxy wrapping slightly easier.

File: tools/snippy/macros.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<macros>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="3.2">snippy</requirement>
```
Comment (bgruening) [inline tools/snippy/macros.xml]: should this one also use the macro @VERSION@

File: tools/snippy/snippy-core.xml [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        Combine multiple Snippy outputs into a core SNP alignment
+    </description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+
+        perl "$__tool_directory__/snippy_core_wrapper.pl"
+            "$is_reference"
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: single quotes for everything can looks like a path or a text :)

File: tools/snippy/snippy-core.xml [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+            --indirs '${" ".join(map(str, $indirs))}'
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: the map is imho not needed
here as well `" ".join("'%s'" % $file for $file in $indirs)`

File: tools/snippy/snippy-core.xml [hunk 12]
```diff
@@ -0,0 +1,76 @@
+<tool id="snippy-core" name="snippy-core" version="@VERSION@">
```
Comment (bgruening) [inline tools/snippy/snippy-core.xml]: can we use a `_` inside the ID

File: tools/snippy/snippy.xml [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+  <macros>
+      <import>macros.xml</import>
+  </macros>
+  <expand macro="requirements" />
+  <expand macro="version_command" />
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        #if $ref.is_of_type("fasta")
+            cp "$ref" "foo.fna" &&
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: single quotes

File: tools/snippy/snippy.xml [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+        #end if
+        #if $ref.is_of_type("genbank")
+            cp "$ref" "foo.gbk" &&
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: single quotes

File: tools/snippy/snippy.xml [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+            #if str( $fastq_input.fastq_input_selector ) == "single"
+                --se "$fastq_input.fastq_input1"
+            #end if
+            #if str( $fastq_input.fastq_input_selector ) == "paired_iv"
+                --peil "$fastq_input.fastq_input1"
+            #end if
+
+        &&
+
+        gunzip out/snps.depth.gz
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: gunzip should probably be a requirement as well ...

File: tools/snippy/snippy.xml [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+                --ref "foo.fna"
+            #end if
+            #if $ref.is_of_type("genbank")
+                --ref "foo.gbk"
+            #end if
+            --mapqual $adv.mapqual
+            --mincov $adv.mincov
+            --minfrac $adv.minfrac
+            #if $adv.rgid
+                --rgid $advanced.rgid
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: single quote

File: tools/snippy/snippy.xml [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+            #end if
+            #if $adv.bwaopt
+                --bwaopt "$advanced.bwaopt"
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: singel quote

File: tools/snippy/snippy.xml (hunk 15 above)
Comment (Slugger70) [reply tools/snippy/snippy.xml]: Gunzip is included in snippy bioconda recipe.

**Comment by bgruening** [comment]
Thanks @Slugger70!

**Comment by lparsons** [comment]
Not sure why, but I'm getting an error when I run this, looks like a problem with a missing dep in either this or the bioconda recipe (depends on your philosophy):
```
### vcf-consensus snps.vcf.gz < reference/ref.fa > snps.consensus.fa

Can't locate Vcf.pm in @INC (you may need to install the Vcf module) (@INC contains: /galaxy/galaxy-app/tool-dependencies/_conda/envs/__snippy@3.2/lib/perl5/site_perl/5.22.0/x86_64-linux-thread-multi /galaxy/galaxy-app/tool-dependencies/_conda/envs/__snippy@3.2/lib/perl5/site_perl/5.22.0 /galaxy/galaxy-app/tool-dependencies/_conda/envs/__snippy@3.2/lib/perl5/5.22.0/x86_64-linux-thread-multi /galaxy/galaxy-app/tool-dependencies/_conda/envs/__snippy@3.2/lib/perl5/5.22.0 .) at /usr/local/bin/vcf-consensus line 6.
BEGIN failed--compilation aborted at /usr/local/bin/vcf-consensus line 6.
```

**Comment by bgruening** [comment]
@Slugger70 can you try again. Its working on usegalaxy.eu

**Comment by bgruening** [comment]
Or do you execute it with some special parameters?

**Comment by Slugger70** [comment]
Hi @lparsons  I'm investigating it now. Seems to work ok for me. Trying to replicate it on a brand new vanilla linux.

**Pull request merged**